hand-written.md

```diff
@@ -1,39 +1,32 @@
-We are pleased to announce the availability of Scala 2.12.0-M3!
+We are happy to announce the availability of Scala 2.12.0-M4, which marks feature completeness for 2.12!
 
-Significant changes since M2 include:
+Traits [#5003](https://github.com/scala/scala/pull/5003) and functions are compiled to their Java 8 equivalents, and we treat Single Abstract Method types and Scala's builtin function types uniformly from type checking to the backend [#4971](https://github.com/scala/scala/pull/4971). We also use invokedynamic for a more natural encoding of other language features [#4896](https://github.com/scala/scala/pull/4896). We've standardized on the GenBCode back-end ([#4814](https://github.com/scala/scala/pull/4814), [#4838](https://github.com/scala/scala/pull/4838)) and the flat classpath implementation is now the default ([#5057](https://github.com/scala/scala/pull/5057)). The optimizer has been completely overhauled for 2.12. This milestone adds box/unbox optimization ([#4858](https://github.com/scala/scala/pull/4858)).
```

Comment:
in "The optimizer has been completely overhauled", maybe "rebuilt" instead of "overhauled"